libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -444,19 +473,36 @@ async def _agenerate(
             # set param to `functions` until core tool/function calling implemented
             raw_tools = params.pop("functions") if "functions" in params else None
             tools = _format_tools_to_vertex_tool(raw_tools) if raw_tools else None
-            response = await chat.send_message_async(
-                message,
-                generation_config=params,
-                tools=tools,
-                safety_settings=safety_settings,
-            )
-            generations = [
-                ChatGeneration(
-                    message=_parse_response_candidate(c),
-                    generation_info=get_generation_info(c, self._is_gemini_model),
+            try:
+                response = chat.send_message_async(
+                    message,
+                    generation_config=params,
+                    tools=tools,
+                    safety_settings=safety_settings,
                 )
-                for c in response.candidates
-            ]
+            except ResponseBlockedError as e:
+                response = e.responses[0]
+            generations = []
+            for c in response.candidates:
+                if c.finish_reason in [
+                    gapic_content_types.Candidate.FinishReason.STOP,
+                    gapic_content_types.Candidate.FinishReason.FINISH_REASON_UNSPECIFIED,
+                ]:
+                    generations.append(
+                        ChatGeneration(
+                            message=_parse_response_candidate(c),
+                            generation_info=get_generation_info(
+                                c, self._is_gemini_model
+                            ),
+                        )
+                    )
+                else:
+                    error_str = "Text was found that violates the safety policy"
+                    if self.raise_on_blocked:
+                        raise ValueError(error_str)
+                    generations.append(
+                        ChatGeneration(message=AIMessage(content=error_str))
+                    )
```

Comment:
please, add an integration test. thanks!

Reply:
It's not obvious how to get the LLM to say something harmful, but I'll check it out.